src/index.js

```diff
@@ -79,20 +84,21 @@ async function main() {
   }
 
 
-  const githubRepositories = await getGithubRepositories(githubUsername);
+  const githubRepositories = await getGithubRepositories(githubUsername, githubToken);
   console.log(`Found ${githubRepositories.length} repositories on github`);
 
   const gitea = {
     url: giteaUrl,
     token: giteaToken,
   };
-
   const giteaUser = await getGiteaUser(gitea);
-  githubRepositories.forEach(
-    async repository => {
+
+  const queue = new PQueue({ concurrency: 4 });
+  await queue.addAll(githubRepositories.map(repository => {
+    return async () => {
       await mirror(repository, gitea, giteaUser);
-    }
-  );
+    };
+  }));
```

Comment:
How about just going one by one? 😄 
This would avoid another dependency and may take a little longer.
something like
```
  for (const repo of githubRepositories) {
    await mirror(... );
  }
```

I think the stability of setting up the mirrors should be more important than the performance.

Reply:
I don't really mind, whatever you like. I just migrated 500 repos, so it was worth it for me, but I can see that this is not an average use case.

Reply:
What was your use-case? Are you running this script/container on a continuous basis or did you just do this once?

Reply:
Just once.

Reply:
Ok. I intended this to be used on a regular basis. Thanks for the change, I will test the changes and merge the PR.